src/ble_utils.cpp

```diff
@@ -1,219 +1,196 @@
+                                                                                                                      .read = _read_cb,
+                                                                                                                      .write = _write_cb,
+                                                                                                                      .user_data = this,
+                                                                                                                      .handle = 0,
+                                                                                                                      .perm = perm}),
+                                                                                                        m_gatt_chrc({.uuid = uuid,
+                                                                                                                     .value_handle = 0,
+                                                                                                                     .properties = props}),
+                                                                                                        m_ccc_enable(ccc_enable)
+    {
+    }
+    ssize_t Characteristic::_read_cb(struct bt_conn *conn,
+                                     const struct bt_gatt_attr *attr,
+                                     void *buf, uint16_t len,
+                                     uint16_t offset)
+    {
+        ARG_UNUSED(conn);
+        ARG_UNUSED(offset);
+        auto instance = static_cast<Characteristic *>(attr->user_data);
+        return instance->read_cb(buf, len);
+    }
+    ssize_t Characteristic::_write_cb(struct bt_conn *conn,
+                                      const struct bt_gatt_attr *attr,
+                                      const void *buf,
+                                      uint16_t len,
+                                      uint16_t offset,
+                                      uint8_t flags)
+    {
+        ARG_UNUSED(conn);
+        ARG_UNUSED(offset);
+        ARG_UNUSED(flags);
+        auto instance = static_cast<Characteristic *>(attr->user_data);
+        return instance->write_cb(buf, len);
+    }
+
+    Service::Service(const bt_uuid *uuid) : m_gatt_service(
+                                                {.attrs = attrs,
+                                                 .attr_count = SVC_ATTR_SIZE,
+                                                 .node = {nullptr}})
+    {
+        const bt_gatt_attr svc_attr =
+            {
+                .uuid = static_cast<const bt_uuid *>(static_cast<const void *>(&uuid::PRIMARY_SVC)),
+                .read = bt_gatt_attr_read_service,
                 .write = nullptr,
-                .user_data = const_cast<bt_gatt_chrc *>(&m_gatt_chrc),
+                .user_data = static_cast<void *>(const_cast<bt_uuid *>(uuid)),
```

Comment:
I think this will not work. The reason of having m_gatt_chrc as part of the user data is that this is how zephyr expects the gatt attribute to be declared. Changing this will probably not work

Reply:
ok nvm, I think this is due to the changes in code format you did. Hence some lines are not at the same position and it is difficult to review on github.